backend/src/app/core/data/dto/filter.py

```diff
@@ -1,18 +1,114 @@
 from enum import Enum
-from typing import List, Union
+from typing import List, Optional, Union
 
 from pydantic import BaseModel
 from sqlalchemy import Column, and_, or_
+from sqlalchemy.orm import Session
 
+from app.core.data.crud.project_metadata import crud_project_meta
+from app.core.data.dto.project_metadata import ProjectMetadataRead
 from app.core.data.orm.code import CodeORM
 from app.core.data.orm.document_tag import DocumentTagORM
 from app.core.data.orm.memo import MemoORM
 from app.core.data.orm.source_document import SourceDocumentORM
+from app.core.data.orm.source_document_data import SourceDocumentDataORM
+from app.core.data.orm.source_document_metadata import SourceDocumentMetadataORM
+from app.core.data.orm.span_annotation import SpanAnnotationORM
 from app.core.data.orm.span_text import SpanTextORM
+from app.core.data.orm.user import UserORM
+
+
+class AggregatedColumn(str, Enum):
+    DOCUMENT_TAG_IDS_LIST = "tag_ids"
+    CODE_IDS_LIST = "code_ids"
+    USER_IDS_LIST = "user_ids"
+    SPAN_ANNOTATION_IDS_LIST = "span_annotation_ids"
+    SPAN_ANNOTATIONS = "span_annotation_tuples"
+
+
+# I believe it is important that key and value are identical!
+class DBColumns(Enum):
+    SPAN_TEXT = "SPAN_TEXT"
```

Comment:
In this file we use both upper- and lowercase values for enums, I think we should stick to one style.